shell/platform/cpp/client_wrapper/include/flutter/json_method_codec.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_JSON_METHOD_CODEC_H_
+#define FLUTTER_SHELL_PLATFORM_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_JSON_METHOD_CODEC_H_
+
+#include "json_type.h"
+#include "method_call.h"
+#include "method_codec.h"
+
+namespace flutter {
+
+// An implementation of MethodCodec that uses JSON strings as the serialization.
+class JsonMethodCodec : public MethodCodec<JsonValueType> {
+ public:
+  // Returns the shared instance of the codec.
+  static const JsonMethodCodec& GetInstance();
+
+  ~JsonMethodCodec() = default;
+
+  // Prevent copying.
+  JsonMethodCodec(JsonMethodCodec const&) = delete;
+  JsonMethodCodec& operator=(JsonMethodCodec const&) = delete;
+
+ protected:
+  // Instances should be obtained via GetInstance.
+  JsonMethodCodec() = default;
+
+  // MethodCodec:
+  std::unique_ptr<MethodCall<JsonValueType>> DecodeMethodCallInternal(
```

Comment:
By convention, every overridden method name has a comment right above it listing the superclass that specifies the virtual method. See files [like this as an example](https://github.com/flutter/engine/blob/cae0836576df83c7d75432226a81f16eb76b285e/shell/common/shell.h#L119). Also a single empty newline after each method definition.